gems/aws-sdk-core/lib/aws-sdk-core/plugins/recursion_detection.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Aws
+  module Plugins
+    # @api private
+    class RecursionDetection < Seahorse::Client::Plugin
```

Comment:
Should the naming be more specific? If this is specifically for Lambda usage, maybe include it in the naming?

Reply:
Eventually this may apply outside of just Lambda.